components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java

```diff
@@ -62,17 +63,27 @@
     private char[] clientSecret;
 
     public String getClientId() {
-        return clientId;
+        if (clientId != null) {
+            return clientId;
+        } else {
+            ServiceHolder.setOidcOauthManager(new OIDCOauthManager());
+            return ServiceHolder.getOidcOauthManager().getClientId();
+        }
     }
 
     public String getClientSecret() {
         return String.valueOf(this.clientSecret);
     }
 
-    public OIDCOauthManager() throws OIDCProviderException {
-        JSONObject clientJson = getClientCredentials();
-        this.clientId = clientJson.getString(Constants.CLIENT_ID_TXT);
-        this.clientSecret = clientJson.getString(Constants.CLIENT_SECRET_TXT).toCharArray();
+    public OIDCOauthManager() {
+        try {
+            JSONObject clientJson = getClientCredentials();
+            this.clientId = clientJson.getString(Constants.CLIENT_ID_TXT);
+            this.clientSecret = clientJson.getString(Constants.CLIENT_SECRET_TXT).toCharArray();
+        } catch (OIDCProviderException e) {
+            log.warn("Registering Client failed due to IDP not available for connection." +
+                    " DCR will be performed while user logs into the Observability Portal.");
+        }
```

Comment:
Suggested text for the warn long: DCR failure due to IDP unavailability, will be re-attempted when a user logs in.

Reply:
Updated as per the comment.